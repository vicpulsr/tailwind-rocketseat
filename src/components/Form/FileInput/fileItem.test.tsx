import { render, screen } from "@testing-library/react";
import FileItem from "./FileItem";
describe("Component: FileItem", () => {
  beforeEach(async () => {
    const formatBytes = jest.fn();
    await formatBytes();
  });

  it("renders correctly with progress error", async () => {
    render(<FileItem name="Nome do arquivo" state="error" size={200} />);

    const textError = "Upload failed, please try again.";
    const buttonErrorId = "btn-error-file-item";
    const containerId = "container-file-item";
    const classContainerWhenError = "bg-error-25";

    expect(screen.getByText(textError)).toBeInTheDocument();
    expect(screen.getByTestId(buttonErrorId)).toBeInTheDocument();
    expect(screen.getByTestId(containerId)).toHaveClass(
      classContainerWhenError,
    );
  });

  it("renders correctly with progress progress", async () => {
    render(<FileItem name="Nome do arquivo" state="progress" size={200} />);

    const textProgress = "80%";

    expect(screen.getByText(textProgress)).toBeInTheDocument();
  });

  it("renders correctly with progress complete", async () => {
    render(<FileItem name="Nome do arquivo" state="complete" size={200} />);

    const textComplete = "100%";
    const iconComplete = "icon-complete-file-item";
    const containerId = "container-file-item";
    const classContainerWhenComplete = "border-violet-500";

    expect(screen.getByText(textComplete)).toBeInTheDocument();
    expect(screen.getByTestId(iconComplete)).toBeInTheDocument();
    expect(screen.getByTestId(containerId)).toHaveClass(
      classContainerWhenComplete,
    );
  });

  it('renders correctly with progress state', () => {
    const { container } = render(<FileItem state="progress" name="Nome do arquivo" size={1024} />);
    expect(container).toMatchSnapshot();
  });
});
