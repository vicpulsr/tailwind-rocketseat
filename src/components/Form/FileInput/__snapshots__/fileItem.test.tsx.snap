// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FileItem renders correctly with progress state 1`] = `
<div>
  <div
    class="group flex items-start gap-4 rounded-lg border border-zinc-200 p-4 dark:border-zinc-700"
    data-testid="container-file-item"
  >
    <div
      class="rounded-full border-4 border-violet-100 bg-violet-200 p-2 text-violet-600 dark:border-zinc-700 dark:bg-zinc-800 dark:text-zinc-500"
    >
      <svg
        class="lucide lucide-cloud-upload size-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 14.899A7 7 0 1 1 15.71 8h1.79a4.5 4.5 0 0 1 2.5 8.242"
        />
        <path
          d="M12 12v9"
        />
        <path
          d="m16 16-4-4-4 4"
        />
      </svg>
    </div>
    <div
      class="flex flex-1 flex-col items-start gap-1"
    >
      <div
        class="flex flex-col"
      >
        <span
          class="text-sm font-medium text-zinc-700 dark:text-zinc-100"
        >
          Nome do arquivo
        </span>
        <span
          class="text-sm text-zinc-500 dark:text-zinc-400"
        >
          1.0 KB
        </span>
      </div>
      <div
        class="flex w-full items-center gap-3"
      >
        <div
          class="h-2 flex-1 rounded-full bg-zinc-100 dark:bg-zinc-600"
        >
          <div
            class="h-2 rounded-full bg-violet-600 dark:bg-violet-400"
            style="width: 80%;"
          />
        </div>
        <span
          class="text-sm font-medium text-zinc-700 dark:text-zinc-300"
        >
          80%
        </span>
      </div>
    </div>
    <button
      class="py-2 text-sm font-semibold outline-none focus-visible:ring-2 focus-visibile:ring-violet-500 focus-visible:ring-offset-2 active:opacity-80 rounded-md px-2 shadow-none text-zinc-500 hover:bg-zinc-50 dark:hover:bg-white/5 dark:text-zinc-400"
      type="button"
    >
      <svg
        class="lucide lucide-trash2 size-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 6h18"
        />
        <path
          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
        />
        <path
          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
        />
        <line
          x1="10"
          x2="10"
          y1="11"
          y2="17"
        />
        <line
          x1="14"
          x2="14"
          y1="11"
          y2="17"
        />
      </svg>
    </button>
  </div>
</div>
`;
